### PR TITLE
Support freezing of large trees

### PR DESCRIPTION
**Pull request by oschwald**
Previously, `write()` would fail on Linux when the buffer was greater than 2,147,479,552 bytes:

> On Linux, write() (and similar system calls) will transfer at most 0x7ffff000 (2,147,479,552) bytes, returning the number of bytes actually  transferred.  (This is true on both 32-bit and 64-bit systems.)

I also took the opportunity to switch this repo to use `clang-format` instead of uncrustify.

### CODE REVIEW

File: c/tree.c [hunk 1]
```diff
@@ -1467,31 +1467,27 @@ LOCAL void assign_node_number(MMDBW_tree_s *tree, MMDBW_node_s *node,
 void freeze_tree(MMDBW_tree_s *tree, char *filename, char *frozen_params,
                  size_t frozen_params_size)
 {
-#ifdef WIN32
-    int fd = open(filename, O_CREAT | O_TRUNC | O_RDWR);
-#else
-    int fd = open(filename, O_CREAT | O_TRUNC | O_RDWR, (mode_t)0644);
-#endif
-    if (fd == -1) {
+    FILE *file = fopen(filename, "wb");
```
Comment (horgh) [inline c/tree.c]: Interesting that it was opened for reading as well as writing previously.

Comment (oschwald) [reply c/tree.c]: Yeah, I am not sure what that was about.

File: c/tree.c [hunk 2]
```diff
@@ -1467,31 +1467,27 @@ LOCAL void assign_node_number(MMDBW_tree_s *tree, MMDBW_node_s *node,
...
-    freeze_to_fd(&args, SEVENTEEN_NULLS, 17);
-    freeze_to_fd(&args, FREEZE_SEPARATOR, FREEZE_SEPARATOR_LENGTH);
+    freeze_to_file(&args, SEVENTEEN_NULLS, 17);
+    freeze_to_file(&args, FREEZE_SEPARATOR, FREEZE_SEPARATOR_LENGTH);
 
-    freeze_data_to_fd(fd, tree);
+    freeze_data_to_file(&args, tree);
 
-    if (close(fd) == -1) {
+    if (fclose(file) == -1) {
```
[the remark this answers is not in the document]
Comment (oschwald) [reply c/tree.c]: Doh, thanks!

File: c/tree.c [hunk 3]
```diff
@@ -2211,17 +2194,17 @@ LOCAL void *checked_malloc(size_t size)
     return ptr;
 }
 
-LOCAL void checked_write(int fd, char *filename, void *buffer,
-                         ssize_t count)
+LOCAL void checked_fwrite(FILE *file, char *filename, void *buffer,
+                          ssize_t count)
```
[the remark this answers is not in the document]
Comment (oschwald) [reply c/tree.c]: Weird, `write()` also expects a `size_t`. I am not sure why we were using a `ssize_t`.

**Pull request merged**